src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputAstRegressionMethodDefArrayDeclarator.txt

```diff
@@ -0,0 +1,138 @@
+PACKAGE_DEF -> package [1:0]
+|--ANNOTATIONS -> ANNOTATIONS [1:39]
+|--DOT -> . [1:39]
+|   |--DOT -> . [1:28]
+|   |   |--DOT -> . [1:22]
+|   |   |   |--DOT -> . [1:11]
+|   |   |   |   |--IDENT -> com [1:8]
+|   |   |   |   `--IDENT -> puppycrawl [1:12]
+|   |   |   `--IDENT -> tools [1:23]
+|   |   `--IDENT -> checkstyle [1:29]
+|   `--IDENT -> grammar [1:40]
+`--SEMI -> ; [1:47]
+CLASS_DEF -> CLASS_DEF [3:0]
+|--MODIFIERS -> MODIFIERS [3:0]
+|   `--LITERAL_PUBLIC -> public [3:0]
+|--LITERAL_CLASS -> class [3:7]
+|--IDENT -> InputAstRegressionMethodDefArrayDeclarator [3:13]
+`--OBJBLOCK -> OBJBLOCK [3:56]
+    |--LCURLY -> { [3:56]
+    |--METHOD_DEF -> METHOD_DEF [4:4]
+    |   |--MODIFIERS -> MODIFIERS [4:4]
+    |   |--TYPE -> TYPE [4:4]
+    |   |   |--LITERAL_DOUBLE -> double [4:4]
+    |   |   `--ARRAY_DECLARATOR -> [ [4:10]
+    |   |       `--RBRACK -> ] [4:11]
+    |   |--IDENT -> computeDerivatives [4:13]
+    |   |--LPAREN -> ( [4:31]
+    |   |--PARAMETERS -> PARAMETERS [4:32]
+    |   |--RPAREN -> ) [4:32]
+    |   `--SLIST -> { [4:34]
+    |       |--LITERAL_RETURN -> return [5:8]
+    |       |   |--EXPR -> EXPR [5:15]
+    |       |   |   `--LITERAL_NULL -> null [5:15]
+    |       |   `--SEMI -> ; [5:19]
+    |       `--RCURLY -> } [6:4]
+    |--METHOD_DEF -> METHOD_DEF [8:4]
+    |   |--MODIFIERS -> MODIFIERS [8:4]
+    |   |--TYPE -> TYPE [8:4]
+    |   |   `--LITERAL_DOUBLE -> double [8:4]
+    |   |--IDENT -> computeIntegrations [8:11]
+    |   |--LPAREN -> ( [8:30]
+    |   |--PARAMETERS -> PARAMETERS [8:31]
+    |   |--RPAREN -> ) [8:31]
+    |   `--SLIST -> { [8:33]
+    |       |--LITERAL_RETURN -> return [9:8]
+    |       |   |--EXPR -> EXPR [9:15]
+    |       |   |   `--NUM_FLOAT -> 1.1 [9:15]
+    |       |   `--SEMI -> ; [9:18]
+    |       `--RCURLY -> } [10:4]
+    |--METHOD_DEF -> METHOD_DEF [12:4]
+    |   |--MODIFIERS -> MODIFIERS [12:4]
+    |   |--TYPE -> TYPE [12:4]
+    |   |   |--LITERAL_DOUBLE -> double [12:4]
+    |   |   `--ARRAY_DECLARATOR -> [ [15:12]
+    |   |       |--ARRAY_DECLARATOR -> [ [14:12]
```

Comment:
we need to update token javadoc
https://github.com/checkstyle/checkstyle/blob/578a97541eb50f27db3f5d8f4d9b16d09d8618c6/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java#L531

Reply:
```bash
➜  src cat Test.java
public class Test {
    int[] x;
}

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.43-SNAPSHOT-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |--TYPE -> TYPE [2:4]
    |   |   |--LITERAL_INT -> int [2:4]
    |   |   `--ARRAY_DECLARATOR -> [ [2:7]
    |   |       `--RBRACK -> ] [2:8]
    |   |--IDENT -> x [2:10]
    |   `--SEMI -> ; [2:11]
    `--RCURLY -> } [3:0]

```

Result:
![image](https://user-images.githubusercontent.com/31252532/120135057-6bf24d80-c19d-11eb-9043-e855aa303fde.png)

Reply:
Done.

Reply:
there are few places 
one of them
https://github.com/checkstyle/checkstyle/blob/3c0f6ab9e066d61eb09d232849c5d7fa2f216c85/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java#L957-L959

please recheck/fix all.

Reply:
```bash
➜  src cat Test.java
public class Test {
    int[] y =
            {
                    1,
                    2,
            };
}

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.44-SNAPSHOT-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |--TYPE -> TYPE [2:4]
    |   |   |--LITERAL_INT -> int [2:4]
    |   |   `--ARRAY_DECLARATOR -> [ [2:7]
    |   |       `--RBRACK -> ] [2:8]
    |   |--IDENT -> y [2:10]
    |   |--ASSIGN -> = [2:12]
    |   |   `--ARRAY_INIT -> { [3:12]
    |   |       |--EXPR -> EXPR [4:20]
    |   |       |   `--NUM_INT -> 1 [4:20]
    |   |       |--COMMA -> , [4:21]
    |   |       |--EXPR -> EXPR [5:20]
    |   |       |   `--NUM_INT -> 2 [5:20]
    |   |       |--COMMA -> , [5:21]
    |   |       `--RCURLY -> } [6:12]
    |   `--SEMI -> ; [6:13]
    `--RCURLY -> } [7:0]

````

```bash

➜  src cat Test.java
public class Test {
    int[] z = new int[]
            {
                    1,
                    2,
            };
}

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.44-SNAPSHOT-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |--TYPE -> TYPE [2:4]
    |   |   |--LITERAL_INT -> int [2:4]
    |   |   `--ARRAY_DECLARATOR -> [ [2:7]
    |   |       `--RBRACK -> ] [2:8]
    |   |--IDENT -> z [2:10]
    |   |--ASSIGN -> = [2:12]
    |   |   `--EXPR -> EXPR [2:14]
    |   |       `--LITERAL_NEW -> new [2:14]
    |   |           |--LITERAL_INT -> int [2:18]
    |   |           |--ARRAY_DECLARATOR -> [ [2:21]
    |   |           |   `--RBRACK -> ] [2:22]
    |   |           `--ARRAY_INIT -> { [3:12]
    |   |               |--EXPR -> EXPR [4:20]
    |   |               |   `--NUM_INT -> 1 [4:20]
    |   |               |--COMMA -> , [4:21]
    |   |               |--EXPR -> EXPR [5:20]
    |   |               |   `--NUM_INT -> 2 [5:20]
    |   |               |--COMMA -> , [5:21]
    |   |               `--RCURLY -> } [6:12]
    |   `--SEMI -> ; [6:13]
    `--RCURLY -> } [7:0]


```

Found a problem:

```bash
➜  src cat Test.java
public class Test {
    int a[];
}

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.44-SNAPSHOT-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [2:9]
    |   |--MODIFIERS -> MODIFIERS [2:9]
    |   |--TYPE -> TYPE [2:9]
    |   |   `--ARRAY_DECLARATOR -> [ [2:9]
    |   |       |--LITERAL_INT -> int [2:4]
    |   |       `--RBRACK -> ] [2:10]
    |   |--IDENT -> a [2:8]
    |   `--SEMI -> ; [2:11]
    `--RCURLY -> } [3:0]

```

C-style array declarations still have the issue. I am on it.

Reply:
Fixed now:

```bash
➜  src cat Test.java 
public class Test {
    int a[];
}

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.44-SNAPSHOT-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--VARIABLE_DEF -> VARIABLE_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |--TYPE -> TYPE [2:4]
    |   |   |--LITERAL_INT -> int [2:4]
    |   |   `--ARRAY_DECLARATOR -> [ [2:9]
    |   |       `--RBRACK -> ] [2:10]
    |   |--IDENT -> a [2:8]
    |   `--SEMI -> ; [2:11]
    `--RCURLY -> } [3:0]



```